### PR TITLE
feat: basic toc functionality

### PR DESCRIPTION
**Pull request by TomWoodward**
i still have some more functionality from the issue to work on, but this covers the basics.

### CODE REVIEW

File: src/app/content/components/Sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+  ${(props) => !props.open && css`
+    span {
+      margin-top: 15%;
+    }
+  `}
+
+  ${(props) => props.open && css`
+    span:nth-child(1) {
+      transform: rotateZ(45deg);
```
Comment (philschatz) [inline src/app/content/components/Sidebar.tsx]: Ooh, nifty transition!

File: src/app/content/components/Sidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+  public render() {
+    const {open, book, closeToc, openToc} = this.props;
+    return <SidebarPlaceholder open={open}>
+      <SidebarBody open={open}>
+        <SidebarControl
+          open={open}
+          onClick={() => open ? closeToc() : openToc()}
+          role='button'
+          aria-label={`Click to ${open ? 'close' : 'open'} the Table of Contents`}
```
Comment (philschatz) [inline src/app/content/components/Sidebar.tsx]: Hmm, maybe it's time to open an i18n PR...

File: src/app/content/guards.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { ArchiveTree, ArchiveTreeSection } from './types';
+
```
Comment (philschatz) [inline src/app/content/guards.ts]: Ooh, good idea. One place for all the type-guards

**Review by philschatz — DISMISSED** [review]

File: src/app/content/components/Sidebar.tsx (hunk 2 above)
Comment (TomWoodward) [reply src/app/content/components/Sidebar.tsx]: yea i would prefer to do that before we dig our hole much bigger

**Review by philschatz — APPROVED** [review]

**Pull request merged**